api/src/main/java/io/minio/PostPolicy.java

```diff
@@ -187,7 +217,7 @@ public String bucketName() {
     formData.put("x-amz-algorithm", ALGORITHM);
 
     DateTime date = new DateTime();
-    String region = BucketRegionCache.INSTANCE.region(this.bucketName);
+    String region = this.region != null ? this.region : BucketRegionCache.INSTANCE.region(this.bucketName);
```

Comment:
Introducing region setter would confuse existing users as nowhere in minio-java we ask region.

To make it compatible, have `formData()` method to be overloaded with another version like
```
public Map<String,String> formData(String accessKey, String secretKey, String region)
public Map<String,String> formData(String accessKey, String secretKey)
```

where the second one uses the first one with `BucketRegionCache`.

Hope this works for you.